magellan-sample-advanced/src/main/java/com/wealthfront/magellan/sample/advanced/SampleApplication.java

```diff
@@ -0,0 +1,20 @@
+package com.wealthfront.magellan.sample.advanced;
+
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.Context;
+
+import dagger.Component;
+
+public class SampleApplication extends Application {
+
+    public static SampleApplication app(Context context) {
+        return (SampleApplication) context.getApplicationContext();
+    }
+
+    public AppComponent injector() {
+        return DaggerAppComponent.create();
```

Comment:
Shouldn't this be `DaggerAppComponent.builder().appModule(new AppModule(this)).build();`, also shouldn't this be built only once in `onCreate`?